exporter.go

```diff
@@ -427,26 +399,14 @@ func main() {
 		}
 	})
 
-	listener, err := getListener(*listenAddr)
-	if err != nil {
-		log.Fatalf("Could not create listener: %v", err)
-	}
-
-	if *securedMetrics {
```

Comment:
related to the other comment we can keep this logic and check if a path to the config was provided and use `web.ListenAndServe` instead of `srv.Serve` since the former doesn't support unix sockets.

Reply:
thanks for the comments.
regarding exporter toolkit status please see [https://github.com/prometheus/exporter-toolkit/issues/45 ](url)
so I think experimental part is pretty obsolete by now. (and yes, the package is used in a lot of exporters)

I'll take a look at the other comments.

Reply:
does it make sense if we are doing it like it's now done in the last node_exporter version? not long ago they added socket option directly in the exporter-toolkit.

Reply:
@lucacome can you please take a look at my previous comment? thanks

Reply:
@lucian-vanghele sorry for the long wait, I've finally found some time to work on this.
If you want to take a look at #240 🙂

Reply:
Just realized I linked the wrong PR, it was #420 😅